src/Containers/Microsoft.NET.Build.Containers/Registry.cs

```diff
@@ -55,8 +57,13 @@ internal sealed class Registry
 
     public Registry(Uri baseUri)
     {
-        BaseUri = baseUri;
         RegistryName = DeriveRegistryName(baseUri);
+        BaseUri = baseUri;
+        // "docker.io" is not a real registry. Replace the uri to refer to an actual registry.
+        if (BaseUri.Host == ContainerHelpers.DockerRegistryAlias)
+        {
+            BaseUri = new UriBuilder(BaseUri.ToString()) { Host = DockerHubRegistry1 }.Uri;
+        }
```

Comment:
I've moved the mapping of the `docker.io` alias here, so all log messages contain `docker.io` (when that is what the user specified).